integration-tests/publish.go

```diff
@@ -0,0 +1,105 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"github.com/CESARBR/knot-cloud-storage/pkg/data"
+	"github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/logging"
+	"github.com/google/go-querystring/query"
+	"github.com/streadway/amqp"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"time"
+)
+
+func sendRequest(svc *requestInfo) (*http.Response, error) {
+	values, err := query.Values(svc.options)
+	failOnError(err, "Failed to get query values")
+
+	queryString := "?" + values.Encode()
+	client := &http.Client{Timeout: 10 * time.Second}
+	req, err := http.NewRequest(svc.method, svc.url+queryString, bytes.NewBuffer(svc.data))
+	failOnError(err, "Failed to do a HTTP request")
+
+	req.Header.Set("auth_token", svc.authorization)
+	req.Header.Set("Content-Type", svc.contentType)
+
+	return client.Do(req)
+}
+
+
+func publishToken(userTokenValue string, actualTime time.Time) {
```

Comment:
I suggest you always think about when new tests will be added. Is this code easy to extend? For example, I think `publishToken` should become a generic function, used by the test functions that need it.

Reply:
Done. I set the `usrToken string` and `message_body []byte` and renamed the function to `publishData`.